### PR TITLE
CNS-834: added allowed buyers list to plans

### PR DESCRIPTION
**Pull request by oren-lava**
In this PR I added an allowed buyers list to plans. Only the addresses that are on the list can buy/renew/upgrade the plan's subscription. If the allowed buyers list is empty, anyone can buy the plan.

Note, there is no need for migrator since a nil array is not breaking the code.

### CODE REVIEW

File: x/subscription/keeper/subscription_test.go [hunk 1]
```diff
@@ -2432,3 +2432,132 @@ func TestProjectsNumEnforcement(t *testing.T) {
...
+	require.NoError(t, err)
+
+	// change the premium plan's allowed buyers list - only consumer2 can purchase now
+	premiumPlan := ts.Plan("premium")
+	premiumPlan.AllowedBuyers = append(premiumPlan.AllowedBuyers, consumer2)
+	err = ts.TxProposalAddPlans(premiumPlan)
+	require.NoError(t, err)
+
+	// try buying a future subscription for the premium plan -> should fail
+	_, err = ts.TxSubscriptionBuy(consumer1, consumer1, premiumPlan.Index, 1, false, true)
```
Comment (Yaroms) [inline x/subscription/keeper/subscription_test.go:2535]: add test with the allowed consumer

File: x/subscription/keeper/subscription_test.go [hunk 2]
```diff
@@ -2432,3 +2432,132 @@ func TestProjectsNumEnforcement(t *testing.T) {
...
+// Scenarios:
+// 1. try to upgrade subscription to a plan the user is not allowed to purchase -> should fail
+func TestAllowedBuyersUpgradeSubscription(t *testing.T) {
+	ts := newTester(t)
+	ts.SetupAccounts(2, 0, 0) // 2 sub, 0 adm, 0 dev
+	_, consumer1 := ts.Account("sub1")
+	_, consumer2 := ts.Account("sub2")
+
+	// buy subscription with empty allowed buyers list (without auto-renewal)
+	freePlan := ts.Plan("free")
```
Comment (Yaroms) [inline x/subscription/keeper/subscription_test.go:2549]: check that it works for cosumer2

File: x/subscription/keeper/subscription_test.go (hunk 1 above)
Comment (oren-lava) [reply x/subscription/keeper/subscription_test.go:2535]: done. see https://github.com/lavanet/lava/commit/c35bb71ec4ea555c4e7ec9670cdc91007255c476

File: x/subscription/keeper/subscription_test.go (hunk 2 above)
Comment (oren-lava) [reply x/subscription/keeper/subscription_test.go:2555]: done. see https://github.com/lavanet/lava/commit/c35bb71ec4ea555c4e7ec9670cdc91007255c476

**Review by Yaroms — APPROVED** [review]

**Pull request merged**